server/src/main/java/keywhiz/service/providers/ClientAuthFactory.java

```diff
@@ -134,12 +135,23 @@ public Client provide(ContainerRequest containerRequest,
   }
 
   private Client authorizeClientFromXfccHeader(XfccSourceConfig xfccConfig,
-      List<String> xfccHeaderValues, Principal requestPrincipal) {
+      List<String> xfccHeaderValues, Principal requestPrincipal,
+      ContainerRequest containerRequest) {
     // Do not allow the XFCC header to be set by all incoming traffic. This throws a
     // NotAuthorizedException when the traffic is not coming from a source allowed to set the
     // header.
     validateXfccHeaderAllowed(xfccConfig, requestPrincipal);
 
+    Optional<String> spiffeIdHeader = Optional.ofNullable(xfccConfig.callerSpiffeIdHeader());
+    Optional<URI> callerSpiffeId = spiffeIdHeader.isPresent()
```

Comment:
nit: You can use [Optional.flatMap()](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#flatMap-java.util.function.Function-) for this rather than a ternary:
```
spiffeIdHeader.flatMap(header -> ClientAuthenticator.getSpiffeIdFromHeader(containerRequest, header))
```
will return an Optional which is empty if `spiffeIdHeader` was empty, and otherwise is the value of `getSpiffeIdFromHeader`.